### PR TITLE
feat(kustomize): support envsubst in commonAnnotations and commonLabels

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
Replacement for #11333 PR

### CODE REVIEW

File: util/kustomize/kustomize.go [hunk 1]
```diff
@@ -124,7 +124,11 @@ func (k *kustomize) Build(opts *v1alpha1.ApplicationSourceKustomize, kustomizeOp
 			if opts.ForceCommonLabels {
 				args = append(args, "--force")
 			}
-			cmd := exec.Command(k.getBinaryPath(), append(args, mapToEditAddArgs(opts.CommonLabels)...)...)
+			commonLabels := map[string]string{}
+			for name, value := range opts.CommonLabels {
+				commonLabels[name] = envVars.Envsubst(value)
```
Comment (crenshaw-dev) [inline util/kustomize/kustomize.go:129]: Suppose someone is already using a `commonAnnotation` with a $ in it, e.g. `$ilikedollarsigns`, and they expect it to be interpreted literally... will this cause their annotations to be replaced with empty strings?

Comment (crenshaw-dev) [reply util/kustomize/kustomize.go:129]: We could toggle the behavior on a new field, e.g. `commonLabelsEnvsubst`. I don't like adding a new field, but I really don't wanna mess up folks' labels.

Comment (crenshaw-dev) [reply util/kustomize/kustomize.go:129]: Oh wait, these are labels! They can't have dollar signs anyway: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

Comment (crenshaw-dev) [reply util/kustomize/kustomize.go:129]: Ah, but we would still need a toggle for annotations.

**Comment by gxxglyeyez** [comment]
Oren from Komodor, Would love to see this added ❤️

File: util/kustomize/kustomize.go (hunk 1 above)
Comment (crenshaw-dev) [reply util/kustomize/kustomize.go:129]: @AndrewChubatiuk apologies for the lack of response. I had to prioritize some high-priority stuff for 2.6. Let me target this for 2.7.

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks for your patience @AndrewChubatiuk! Would you mind documenting the new field here? https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/application.yaml

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**